### PR TITLE
Update functional tests to not expect STANDARD storage class

### PR DESCRIPTION
**Pull request by nitisht**
Minio server doesn't respond with storage class as object metadata
response, if the storage class is set to STANDARD. This PR updates
the tests to fail if STANDARD is sent as storage class in response
metadata.
    
Additionally, Minio gateway (except S3 Gateway) currently ignore storage 
class requests and hence don't respond with storage class in response
metadata for GetObjectInfo calls. Updated the functional tests to
ignore if storage class is not present in metadata.

### CODE REVIEW

**Comment by poornas** [comment]
@nitisht , travis failure needs to be fixed.

File: functional_tests.go [hunk 1]
```diff
@@ -5554,8 +5554,8 @@ func testStorageClassMetadataPutObject() {
 		logError(testName, function, args, startTime, "", "PutObject failed", err)
 		return
 	}
-	if !reflect.DeepEqual(metadata, fetchMeta("srcObjectSSClass")) {
-		logError(testName, function, args, startTime, "", "Metadata match failed", err)
+	if reflect.DeepEqual(metadata, fetchMeta("srcObjectSSClass")) {
+		logError(testName, function, args, startTime, "", "Metadata match failed, STANDARD should not be a part of response metadata", err)
```
Comment (poornas) [inline functional_tests.go]: Suggest wording this as ""Metadata match failed, STANDARD storage class should not be a part of response metadata".

Comment (poornas) [inline functional_tests.go]: "Metadata match failed, STANDARD storage class should not be a part of response metadata" would be clearer. Suggest changing this message and on line 5694

Comment (nitisht) [reply functional_tests.go]: done, at both the locations

**Comment by nitisht** [comment]
> travis failure needs to be fixed. 

The storage class related test is expected to fail till https://github.com/minio/minio/pull/5338 is merged. Initially we returned `STANDARD` in response metadata, but this is being changed to comply with S3 behaviour.

**Comment by kannappanr** [comment]
@nitisht Can you reword,  `"Metadata match failed, STANDARD storage class should not be a part of response metadata` ? There is no Metadata mismatch here.

**Comment by nitisht** [comment]
PR is updated @harshavardhana @poornas @kannappanr PTAL

**Pull request merged**